Mage.Sets/src/mage/cards/s/SergeantAtArms.java

```diff
@@ -32,9 +33,9 @@ public SergeantAtArms(UUID ownerId, CardSetInfo setInfo) {
         this.addAbility(new KickerAbility("{2}{W}"));
 
         // When Sergeant-at-Arms enters the battlefield, if it was kicked, create two 1/1 white soldier creature tokens.
-        this.addAbility(new EntersBattlefieldTriggeredAbility(new ConditionalOneShotEffect(
-                new CreateTokenEffect(new SoldierToken(), 2), KickedCondition.ONCE,
-                "if it was kicked, create two 1/1 white Soldier creature tokens.")));
+        this.addAbility(new ConditionalInterveningIfTriggeredAbility(
+                new EntersBattlefieldTriggeredAbility(new CreateTokenEffect(new SoldierToken(), 2)), KickedCondition.ONCE,
+                "if it was kicked, create two 1/1 white Soldier creature tokens."));
```

Comment:
the text gen works differently for the ability rule versus the effect, so you have to include `When {this} enters, ` now